src/app/service/pet/pet.service.go

```diff
@@ -276,3 +271,51 @@ func (s *Service) ChangeView(id string, in *dto.ChangeViewPetRequest) (result *d
 		Success: res.Success,
 	}, nil
 }
+
+func (s *Service) Adopt(petId string, in *dto.AdoptByRequest) (result *dto.AdoptByResponse, err *dto.ResponseErr) {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	res, errRes := s.petClient.AdoptPet(ctx, &petproto.AdoptPetRequest{
+		UserId: in.UserID,
+		PetId:  in.PetID,
+	})
+
+	if errRes != nil {
+		st, _ := status.FromError(errRes)
+		log.Error().
+			Err(errRes).
+			Str("service", "pet").
+			Str("module", "adopt").
+			Msg(st.Message())
+		switch st.Code() {
+		case codes.NotFound:
+			return &dto.AdoptByResponse{
+					Success: false,
+				}, &dto.ResponseErr{
```

Comment:
for error cases where results only have ```Success```, we can return ```nil``` instead of ```&dto.AdoptByResponse{
					Success: false,
				}```

Reply:
I'm confused about why it should return nil. Could you explain a bit more?

Reply:
When nil, gRPC makes every field default value. For bool, it's False. So we use nil just to make code shorter

Reply:
Wait, it's not gRPC. But anyways, it still will be the default value, which is False

Reply:
Wait I might be explaining it wrong. Now that I think of it, the reason we use nil is because once an error occurs, we will never use the response and we will forward that error down the services, eventually reaching the user with some status error